treeherder/webapp/api/performance_data.py

```diff
@@ -676,7 +690,35 @@ def list(self, request):
                 item['repository_name'] = repository_name
 
         serializer = self.get_serializer(self.queryset, many=True)
-        return Response(data=serializer.data)
+        serialized_data = serializer.data
+
+        if no_retriggers:
+            serialized_data = self._filter_out_retriggers(serialized_data)
+
+        return Response(data=serialized_data)
+
+    @staticmethod
+    def _filter_out_retriggers(serialized_data: List[dict]) -> List[dict]:
+        """
+        Removes data points resulted from retriggers
+        """
+
+        for perf_summary in serialized_data:
+            retrigger_idxs, seen_push_id = set(), None
```

Comment:
From what it seems, we assume that the data is always grouped by `push_id` (all retriggers are one after another). Is this the case every time?

Reply:
Good question!

To be more precise, it's ordered by the push's timestamp. This is how the grouping by `push_id` is guaranteed.
This has precision down to the second. Technically, there is the possibility of 2 different pushes to land at the same second & some of the data points be randomly mixed.

I checked the database on production & we didn't experience conflicts like these since the end of 2017. But they happened.

Reply:
It isn't clear to me what `retrigger_idxs` means. Could this be renamed so it's clearer?

Reply:
Sure thing. I've renamed this to `retriggered_jobs` so it's clearer.